src/app/fddev/tiles/fd_benchs.c

```diff
@@ -1,32 +1,337 @@
+/* _GNU_SOURCE for recvmmsg and sendmmsg */
+#define _GNU_SOURCE
+
 #include "../../fdctl/run/tiles/tiles.h"
+#include "../../../waltz/xdp/fd_xsk_aio.h"
+#include "../../../waltz/quic/fd_quic.h"
+#include "../../../waltz/tls/test_tls_helper.h"
 
 #include <linux/unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <string.h>
+#include <unistd.h>
+#include <poll.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <time.h>
+
+
+// TODO replace with config
+#define quic_enabled 1
+
+/* max number of buffers batched for receive */
+#define IO_VEC_CNT 128
+
+
+struct signer_ctx {
+  fd_sha512_t sha512[ 1 ];
+
+  uchar public_key[ 32UL ];
+  uchar private_key[ 32UL ];
+};
+typedef struct signer_ctx signer_ctx_t;
+
+
+static void
+signer( void *        _ctx,
+        uchar         signature[ static 64 ],
+        uchar const   payload[ static 130 ] ) {
+  fd_tls_test_sign_ctx_t * ctx = (fd_tls_test_sign_ctx_t *)_ctx;
+  fd_ed25519_sign( signature, payload, 130UL, ctx->public_key, ctx->private_key, ctx->sha512 );
```

Comment:
I'm not sure about signing on the benchs. The design here is that benchg is scalable and does signing, and benchs just sends on the socket (which is very expensive). If you are mixing those two things it makes the scaling story for the bench topology much more complicated.

Reply:
you think doing it in bencho is a better choice?

Reply:
you know, this signing only occurs during handshaking

Reply:
Oh, hmm. Then after that it's just TLS or whatever?